src/gpu/intel/jit/gemm/include/type.hpp

```diff
@@ -63,17 +64,25 @@ class Type {
     constexpr int components()        const { return 1; }
     constexpr bool isInteger()        const { return uint32_t(val) & 0x100000; }
     constexpr bool isFP()             const { return !isInteger(); }
+    constexpr bool isFP4()            const { return uint32_t(val) & 0x40000000; }
     constexpr bool isInt4()           const { return uint32_t(val) & 0x20000000; }
+    constexpr bool is4Bit()           const { return isInt4() || isFP4(); }
     constexpr bool isInt8()           const { return (val == Type::u8)  || (val == Type::s8);  }
     constexpr bool isInt16()          const { return (val == Type::u16) || (val == Type::s16); }
-    constexpr bool isF8()             const { return (val == Type::bf8) || (val == Type::hf8); }
+    constexpr bool isF8()             const { return (val == Type::bf8) || (val == Type::hf8  || (val == Type::f8_e8m0)); }
+    constexpr bool isF4()             const { return (val == Type::f4_e2m1) ;}
     constexpr bool isSigned()         const { return (uint32_t(val) & 0x110000) != 0x100000; }
-    constexpr int bits()              const { return isInt4() ? 4 : (paddedSize() * 8); }
+    constexpr int bits()              const { return is4Bit() ? 4 : (paddedSize() * 8); }
     constexpr int paddedSize()        const { return (uint32_t(val) >> 8) & 0xFF; }
     int log2Size()                    const { subByteCheck(); return uint32_t(val) & 0xFF; }
     int size()                        const { subByteCheck(); return paddedSize(); }
-    constexpr int perByte()           const { return isInt4() ? 2 : 1; }
-    void subByteCheck()               const { if (isInt4()) stub(); }
+    constexpr int perByte()           const { return is4Bit() ? 2 : 1; }
+    void subByteCheck()               const { if (is4Bit()) stub(); }
+
+// Placeholder for data type unimplemented in HW.
+static ngen::DataType ngen_f4() {
```

Comment:
Since we will likely have e3m0 soon, and e2m1 is the official name of the type in nGEN:
```suggestion
static ngen::DataType ngen_e2m1() {
```

Reply:
replaced all with `ngen_f4_e2m1` to match convention throughout the codebase.